js/gl.js

```diff
@@ -420,13 +420,14 @@ function _webglGet(name_, p, type) {
 var Module;
 var wasm_exports;
 
-function resize(canvas, on_resize) {
+function resize(canvas, on_resize, force = false) {
     var dpr = dpi_scale();
     var displayWidth = canvas.clientWidth * dpr;
     var displayHeight = canvas.clientHeight * dpr;
 
     if (canvas.width != displayWidth ||
-        canvas.height != displayHeight) {
+        canvas.height != displayHeight ||
+        force) {
```

Comment:
I am concerned about this `force` flag. Why do we need to send a resize message if size is still the same?

Reply:
haven't looked at the code so I might be wrong

but perhaps it is a way to update the size if the CSS or similar changes?

Reply:
Maybe I am missing something, but I can't see how any sort of resize can be meaningful if  canvas.width/height is still the same?

Reply:
as far as my tests went

when forcing the canvas to be "fullscreen"
resizing the browser window will change `canvas.clientWidth`, but not `canvas.width`

but when removing `width: 100%` and `height: 100%` from the canvas css
manually updating the canvas width using js (for example by doing `canvas.width += 10;`)
will change both `canvas.width` and `canvas.clientWidth` 
so the condition evaluates to false, and on_resize method doesn't get called

same applies for height